### PR TITLE
Add nida thru dbgap parser

### PR DESCRIPTION
**Pull request by warrenstephens**

### CODE REVIEW

**Review by cschreep — CHANGES_REQUESTED** [review]
Looks good, but I think adding tests for the new functionality would be helpful

**Comment by warrenstephens** [comment]
Looks like Jenkins needs a bump to python3.9 -- to get the new tests to pass.

Is this in the devops/cicd/jenkins/jenkins-master/docker/Dockerfile?  -- which simply has "python3"?

**Comment by cschreep** [comment]
@warrenstephens The docker agent it uses is actually configured in the Jenkins cluster itself. The image in question is "helxplatform/agent-docker:v0.0.6" But I confess I don't actually know where that is defined in order to introspect it or make updates. @cnbennett3 is on vacation but he has more in-depth knowledge about deploying new docker agents

There are a few things we can try to get jenkins running these new tests with a one-off config, I'll tinker a little bit. Worst case scenario we can either wait until we add the new agent or force-merge the code, depending on urgency

**Comment by warrenstephens** [comment]
@cschreep  Is there any chance that I could change the Dockerfile and that "something" would rebuild it automatically when pushed here?

It looks like the right Dockerfile -- I docker pulled the agent-docker:v0.0.6 image, and looked inside.

**Review by alexwaldrop — CHANGES_REQUESTED** [review]
@warrenstephens @cschreep I'm of the opinion we should separate the NIDA logic and the original dbgap logic into two separate parsers. Obviously this functionally works in its current form, but it's not really a sustainable solution. What happens when we come across a new datatype that we want to munge into a dbgap-style input format? Are we going to add more logic into this one megaparser? There are fields ("element_action", "study_action", "data_type") that really are designed to be specific to the input data source; right now all of our NIDA data is being read in as dbgap data types and the data type field wasn't really designed to reflect the input parsing format, but rather what data tab you want the elements to appear under on the UI. All of this is to say, I think we should use the plugin interface we have to de-couple dbgap/NIDA parsers into separate parsers.

**Review by alexwaldrop — APPROVED** [review]

File: src/dug/core/annotate.py
Comment (cschreep) [inline src/dug/core/annotate.py:165]: The outer parentheses and default values in the `get` call are not strictly necessary and these can be simplified as 
```
noMessage = not response.get("message")
statusError = response.get("status") == 'Error'
```

File: src/dug/core/parsers/dbgap_parser.py
Comment (cschreep) [inline src/dug/core/parsers/dbgap_parser.py:18]: When dealing with filesystem paths, you might find python's [pathlib](https://docs.python.org/3/library/pathlib.html) helpful

**Review by cschreep — APPROVED** [review]

File: src/dug/core/annotate.py
Comment (warrenstephens) [reply src/dug/core/annotate.py:165]: It's the influence of Go language creeping in.

**Pull request merged**
